android/src/main/java/com/opentokreactnative/OTSessionManager.java

```diff
@@ -245,13 +245,13 @@ public void run() {
                 FrameLayout mPublisherViewContainer = sharedState.getPublisherViewContainer();
                 Publisher mPublisher = sharedState.getPublisher();
                 Session mSession = sharedState.getSession();
-                mPublisherViewContainer.removeAllViews();
-                mPublisherViewContainer = null;
-                sharedState.setPublisherViewContainer(mPublisherViewContainer);
                 if (mSession != null) {
                     mSession.unpublish(mPublisher);
                 }
-                mPublisher.destroy();                
+                mPublisher.destroy();
+                mPublisherViewContainer.removeAllViews();
+                mPublisherViewContainer = null;
+                sharedState.setPublisherViewContainer(mPublisherViewContainer);             
```

Comment:
shsharedState.setPublisherViewContainer(null) in place of sharedState.setPublisherViewContainer(mPublisherViewContainer)  makes better readability.

Reply:
@JayTokBox improved with the latest commit :)